packages/components/organisms/f-takeawaypay-activation/src/components/TakeawaypayActivation.vue

```diff
@@ -55,12 +68,107 @@ export default {
         homeUrl: {
             type: String,
             required: true
+        },
+        authToken: {
+            type: String,
+            default: ''
         }
     },
     data () {
         return {
-            tenantConfigs
+            tenantConfigs,
+            consumerId: '',
+            consumerEmail: '',
+            activationState: ACTIVATION_STATE_NONE
         };
+    },
+
+    computed: {
+        activationStateComponent () {
+            if (this.activationState === ACTIVATION_STATE_AVAILABLE_LOGGED_IN) {
+                return this.activationLoggedIn;
+            }
+
+            if (this.activationState === ACTIVATION_STATE_AVAILABLE_NOT_LOGGED_IN) {
+                return this.activationNotLoggedIn;
+            }
+
+            if (this.activationState === ACTIVATION_STATE_SUCCEEDED) {
+                return this.activationSucceeded;
+            }
+            return this.activationFailed;
+        },
+
+        activationFailed () {
+            return {
+                name: 'activation-failed',
+                props: {
+                    redirectUrl: this.redirectUrl
+                }
+            };
+        },
+
+        activationSucceeded () {
+            return {
+                name: 'activation-successful',
+                props: {
+                    homeUrl: this.homeUrl
+                }
+            };
+        },
+
+        activationLoggedIn () {
+            return {
+                name: 'activation-logged-in',
+                props: {
+                    loginUrl: this.loginUrl,
+                    registrationUrl: this.registrationUrl,
```

Comment:
Seems to be duplication here both `loginUrl` and `registrationUrl` are used twice, this could be abstracted.

Reply:
@matthewhardern IMHO, it is more clear to have all dependent properties specified separately.

Reply:
I can agree, but then it seems pointless using `component` and `is`. It would be even more readable if you just `v-if` `v-else` the components, but I guess then its personal opinion on where you draw your abstraction line. Happy to leave it as is for now.